boot/zephyr/nrf_cleanup.c

```diff
@@ -62,10 +64,12 @@ static NRF_UARTE_Type *nrf_uarte_to_clean[] = {
 };
 #endif
 
+#if !defined(CONFIG_SOC_SERIES_NRF54HX)
 static void nrf_cleanup_clock(void)
```

Comment:
Is there a Kconfig that indicates the availability of NRF_CLOCK API?
If so, it would be cleaner to use it here.